code/ui/blocks/src/controls/Boolean.stories.tsx

```diff
@@ -1,27 +1,46 @@
-import React, { useState } from 'react';
+import React from 'react';
+import { useArgs } from '@storybook/addons';
+import type { Meta, StoryObj } from '@storybook/react';
 import { BooleanControl } from './Boolean';
 
-export default {
-  title: 'Controls/Boolean',
+const meta = {
   component: BooleanControl,
   tags: ['docsPage'],
-};
-
-const Template = (initialValue?: boolean) => {
-  const [value, setValue] = useState(initialValue);
-  return (
-    <>
-      <BooleanControl name="boolean" value={value} onChange={(newVal) => setValue(newVal)} />
-      <pre>{JSON.stringify(value) || 'undefined'}</pre>
-    </>
-  );
-};
+  parameters: { controls: { include: ['value'] } },
+  render: (args) => {
+    const [, updateArgs] = useArgs();
+    const { value, onChange } = args;
+    return (
+      <>
+        <BooleanControl
+          {...args}
+          name="boolean"
+          onChange={(newValue) => {
+            updateArgs({ value: newValue });
+            onChange?.(newValue);
+          }}
+        />
+        <pre>{JSON.stringify(value) || 'undefined'}</pre>
```

Comment:
1. Hmm tough one. I feel like I don't want to make it too magical. I want it to be obvious for developers how this works. It could be polished a bit though, like adding "Current value:" in front or something. If we made it too special I'd be afraid users would expect that to be a default behavior. But I'm open if you have any specific ideas to how this could be presented better.
2. yeah a decorator is a good idea, based on a parameter in the stories that need it. I'll work on that.

Reply:
I guess in the case of these controls, these stories are more for internal rather than external use anyway, so it may not be a big deal where the extra info ("current value" as you say) is rendered.

Reply:
I see what you mean, that the user could expect the raw value to also show up when using the component.
The funny thing here is that all the values are synced to the controls when changed (both ways), so you could even argue that the controls panel is an addon that already shows the value - albeit in a less noticable way.

Reply:
I think there's something quite useful in seeing the raw JSON-ified version of the value though. That's why we added this in the first place.